src/goto-instrument/contracts/contracts.cpp

```diff
@@ -1482,6 +1544,28 @@ void code_contractst::add_contract_check(
     check.destructive_append(ensures_pair.first);
   }
 
+  // Translate ensures_contract(ptr, contract) clauses to assertions
+  // ASSERT ptr == contract;
+  for(auto &expr : ensures_contract)
+  {
+    auto &obeys_expr = to_function_pointer_obeys_contract_expr(expr);
+    replace_symbolt replace(common_replace);
```

Comment:
No need for a copy. (And now I'm starting to wonder why a very similar (the same?) code exists multiple times.

Reply:
> (And now I'm starting to wonder why a very similar (the same?) code exists multiple times

This is mostly because I'm lazy :). The code is now factored into two separate methods.